library_generation/owlbot/templates/poms/cloud_pom.xml.j2

```diff
@@ -72,16 +74,30 @@
       <groupId>com.google.api.grpc</groupId>
       <artifactId>proto-google-iam-v1</artifactId>
     </dependency>
+{%- if repo in ['java-storage', 'java-spanner', 'java-dataproc'] %}
```

Comment:
`excluded_dependencies` [prevents items from the `existing_modules` list to be added into the _final_ `required_dependencies` list](https://github.com/googleapis/sdk-platform-java/blob/6913db5c64e35ec8925c25a04ebd6d529895c4fe/library_generation/owlbot/src/fix_poms.py#L430-L447). `required_dependencies` is then split into `proto_modules` and `grpc_modules` and passed into `update_cloud_bom`.

Each gRPC module (from `required_dependencies` -> `grpc_modules`) passed into `update_cloud_bom` [is added as a `<scope>test</scope>` dependency if not already existing](https://github.com/googleapis/sdk-platform-java/blob/6913db5c64e35ec8925c25a04ebd6d529895c4fe/library_generation/owlbot/src/fix_poms.py#L114-L116). It does nothing if it's already existing. 

Excluding the dependencies via `.repo-metadata.json` would not cause the dependencies to be removed since this logic is only to _prevent new dependencies from being added to the pom_.

Reply:
Adding a `test_dependencies` config key in repo_metadata would make sense. We can intersect the contents of `test_dependencies` and `required_dependencies` to convert existing poms' dependencies to test-scoped.

I think the effort is small when it comes to `fix_poms.py` (the hardcoded array is now obtained from repo-metadata). 

The other part would be to have a mini-script to create a PR for all libraries in google-cloud-java except java-dataproc.

I guess we can keep adding entries to `.repo-metadata` since we already have java-postprocessing-specific ones.

Conclusion: small effort.

Reply:
We decided to rely on the `monorepo` variable instead, since the only case where it would not compile is for `java-dataproc`, but it was because it needed the _transitive_ dependency `proto-google-iam-v1`, so the correct approach is to also set its targeted dependencies as test-scoped and additionally include `proto-google-iam-v1` (imported [here](https://github.com/googleapis/google-cloud-java/blob/80b26b745b943b4da6f9ad61b8ac99da233f969a/java-dataproc/google-cloud-dataproc/src/main/java/com/google/cloud/dataproc/v1/BatchControllerClient.java#L34-L38)).